scripts/install_backend_python_libs_test.py

```diff
@@ -0,0 +1,555 @@
+                u'webencodings': u'0.5.1'
+            }
+        def mock_get_directory_file_contents():
+            return {
+                u'html5lib': u'1.0.1',
+                u'six': u'1.15.0',
+                u'flask': u'1.0.1',
+                u'redis': u'3.5.3',
+                u'webencodings': u'0.5.1',
+            }
+
+        swap_get_requirements_file_contents = self.swap(
+            install_backend_python_libs, '_get_requirements_file_contents',
+            mock_get_requirements_file_contents)
+        swap_get_directory_file_contents = self.swap(
+            install_backend_python_libs, '_get_third_party_directory_contents',
+            mock_get_directory_file_contents)
+        with swap_get_requirements_file_contents, (
+            swap_get_directory_file_contents):
+            self.assertEqual(
+                {
+                    u'flask': (None, u'1.0.1')
+                },
+                install_backend_python_libs.get_mismatches())
+
+    def test_multiple_discrepancies_returns_correct_mismatches(self):
+        def mock_get_requirements_file_contents():
+            return {
+                u'html5lib': u'1.0.1',
+                u'six': u'1.13.0',
+                u'redis': u'3.5.3',
+                u'webencodings': u'0.5.2',
+                u'bleach': u'3.1.5'
+            }
+        def mock_get_directory_file_contents():
+            return {
+                u'html5lib': u'1.0.1',
+                u'six': u'1.15.0',
+                u'flask': u'1.0.1',
+                u'redis': u'3.5.3',
+                u'webencodings': u'0.5.0'
+            }
+
+        swap_get_requirements_file_contents = self.swap(
+            install_backend_python_libs, '_get_requirements_file_contents',
+            mock_get_requirements_file_contents)
+        swap_get_directory_file_contents = self.swap(
+            install_backend_python_libs, '_get_third_party_directory_contents',
+            mock_get_directory_file_contents)
+        with swap_get_requirements_file_contents, (
+            swap_get_directory_file_contents):
+            self.assertEqual(
+                {
+                    u'flask': (None, u'1.0.1'),
+                    u'six': (u'1.13.0', u'1.15.0'),
+                    u'bleach': (u'3.1.5', None),
+                    u'webencodings': (u'0.5.2', u'0.5.0')
+                },
+                install_backend_python_libs.get_mismatches())
+
+    def test_rectify_party_directory_handles_library_upgrade(self):
+        cmd_token_list = []
+        def mock_check_call(cmd_tokens):
+            cmd_token_list.append(cmd_tokens)
+
+        mismatches = {
+            u'flask': (u'1.1.1', u'1.0.1'),
+            u'six': (u'1.15.0', u'1.13.0')
+        }
+
+        swap_check_call = self.swap(subprocess, 'check_call', mock_check_call)
+        with swap_check_call:
+            install_backend_python_libs._rectify_third_party_directory( # pylint: disable=protected-access
```

Comment:
Don't call private methods in tests. See https://github.com/oppia/oppia/wiki/Writing-backend-tests#common-testing-scenarios for guidance on testing private methods.

Ditto throughout this file.